source/common/access_log/access_log_impl.h

```diff
@@ -305,6 +308,48 @@ class AccessLogFactory {
    */
   static InstanceSharedPtr fromProto(const envoy::config::accesslog::v3::AccessLog& config,
                                      Server::Configuration::CommonFactoryContext& context);
+
+  /**
+   * Template method to create an access log filter from proto configuration for non-HTTP access
+   * loggers.
+   */
+  template <class Context>
+  static FilterBasePtr<Context>
+  accessLogFilterFromProto(const envoy::config::accesslog::v3::AccessLogFilter& config,
+                           Server::Configuration::CommonFactoryContext& context) {
+    if (!config.has_extension_filter()) {
+      ExceptionUtil::throwEnvoyException(
+          "Access log filter: only extension filter is supported by non-HTTP access loggers.");
```

Comment:
it seems the tcp proxy filter's access logger doesn't reject this.

Reply:
I think tcp proxy use the HTTP access logger directly. It's not the non-HTTP access loggers.